code/datums/components/aura_healing.dm

```diff
@@ -0,0 +1,168 @@
+#define HEAL_EFFECT_COOLDOWN (1 SECONDS)
+
+/// Applies healing to those in the area.
+/// Will provide them with an alert while they're in range, as well as
+/// give them a healing particle.
+/// Can be applied to those only with a trait conditionally.
+/atom/movable/screen/alert/aura_healing
+	name = "Aura Healing"
+	icon_state = "template"
+
+/datum/component/aura_healing
+	/// The range of which to heal
+	var/range
+
+	/// Whether or not you must be a visible object of the parent
+	var/requires_visibility = TRUE
+
+	/// Brute damage to heal over a second
+	var/brute_heal = 0
+
+	/// Burn damage to heal over a second
+	var/burn_heal = 0
+
+	/// Toxin damage to heal over a second
+	var/toxin_heal = 0
+
+	/// Suffocation damage to heal over a second
+	var/suffocation_heal = 0
+
+	/// Stamina damage to heal over a second
+	var/stamina_heal = 0
+
+	/// Amount of cloning damage to heal over a second
+	var/clone_heal = 0
+
+	/// Amount of blood to heal over a second
+	var/blood_heal = 0
+
+	/// Map of organ (such as ORGAN_SLOT_BRAIN) to damage heal over a second
+	var/list/organ_healing = null
+
+	/// Shows that aura heals fractures
+	var/heal_fractures = FALSE
+
+	/// Shows that aura heals internal bleedings
+	var/heal_internal_bleedings = FALSE
+
+	/// Showa that aura heals critical burns
+
+	var/heal_critical_burns = FALSE
+
+	/// Amount of damage to heal on simple mobs over a second
+	var/simple_heal = 0
+
+	/// Trait to limit healing to, if set
+	var/limit_to_trait = null
+
+	/// The color to give the healing visual
+	var/healing_color = COLOR_GREEN
+
+	/// A list of being healed to active alerts
+	var/list/current_alerts = list()
+
+	COOLDOWN_DECLARE(last_heal_effect_time)
+
+/datum/component/aura_healing/Initialize(
+	range,
+	requires_visibility = TRUE,
+	brute_heal = 0,
+	burn_heal = 0,
+	toxin_heal = 0,
+	suffocation_heal = 0,
+	stamina_heal = 0,
+	clone_heal = 0,
+	blood_heal = 0,
+	organ_healing = null,
+	heal_fractures = FALSE,
+	heal_internal_bleedings = FALSE,
+	simple_heal = 0,
+	limit_to_trait = null,
+	healing_color = COLOR_GREEN,
+)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	START_PROCESSING(SSaura_healing, src)
+
+	src.range = range
+	src.requires_visibility = requires_visibility
+	src.brute_heal = brute_heal
+	src.burn_heal = burn_heal
+	src.toxin_heal = toxin_heal
+	src.suffocation_heal = suffocation_heal
+	src.stamina_heal = stamina_heal
+	src.clone_heal = clone_heal
+	src.blood_heal = blood_heal
+	src.organ_healing = organ_healing
+	src.heal_fractures = heal_fractures
+	src.heal_internal_bleedings = heal_internal_bleedings
+	src.simple_heal = simple_heal
+	src.limit_to_trait = limit_to_trait
+	src.healing_color = healing_color
```

Comment:
I'd prefer it if you made the arguments start with an underscore, so we don't need the `src.`